### DISCUSSION TITLE
Automation of debugger attach?

### DISCUSSION THREAD

**Discussion by sliakat** [microsoft/vscode-discussions#618] [Extension Development QnA] [answered]
I wanted to ask about the possibility of automating the click that will start VSCode debugger.

Here is the situation: I want to use the Python debugger (debugpy) in attach configuration. From an external process, I have debugpy listening on a port, specified in the launch.json config in the working Python folder. I want that external process to be able to launch VSCode, put me in the correct folder for the Python module I want to debug, pop the module open, and then attach the debugger to it using the existing launch.json in that folder/.vscode.

Using command line arguments, I can have my process fire up VSCode and point me to the correct folder + module. However, I have not seen any path to automating the debugger attach, either through command line tools or through custom tasks.

If such automation is possible, I'd appreciate any guidance in the correct direction.

Thanks!

**Comment by isidorn** [comment, marked as answer]
You could write an extension that achieves this. 
I am not sure if this is possible with the Python extension. These docs should help figure that out https://code.visualstudio.com/docs/python/debugging
